docs/monitoring/docs/021-gs-monitoring-custom-metrics.md

```diff
@@ -0,0 +1,149 @@
+
+```bash
+kubectl label namespace default istio-injection=enabled
+namespace "default" labeled
+```
+
+More details on deploying your application with Istio is [here](https://istio.io/docs/setup/kubernetes/quick-start#deploy-your-application) .
+
+It is needed to add the `sidecar.istio.io/inject` annotation with value `true` to the pod template spec to enable injection as done [here](https://github.com/kyma-project/examples/blob/master/monitoring-custom-metrics/deployment/deployment.yaml#L12).
+
+
+```yaml
+spec:
+  template:
+    metadata:
+      annotations:
+        sidecar.istio.io/inject: "true"
+```
+
+More details on installing the Istio Sidecar is here [here](https://istio.io/docs/setup/kubernetes/sidecar-injection/#policy).
+
+Following ports are used in the pod:
+1. Port: **8080**, as envoy will capture traffic only for ports listed in containerPorts of the pod, `containerPort: 8080`, or the list of ports in the annotation `traffic.sidecar.istio.io/includeInboundPorts`. Thus, this port is a part of the service mesh and can be used for application needs.
+
+2. Port: **8081**, the excluded port from service mesh and is used for exposing metrics only. So the network traffic **bypasses** Envoy and goes straight to the container. In Kyma, use the suggested port `8081` to expose metrics.
+
+### Expose a sample metrics application
+
+In order to expose prometheus metrics in Golang the prometheus community provides a library, [client_golang](https://github.com/prometheus/client_golang).
+
+This example has taken a basic example where `Gauge` and `Counter` metrics are exported using the package `prometheus`.
+
+1. Deploy the sample metrics application
+    ```bash
+    kubectl apply -f https://raw.githubusercontent.com/kyma-project/examples/master/monitoring-custom-metrics/deployment/deployment.yaml
+    
+    kubectl apply -f https://raw.githubusercontent.com/kyma-project/examples/master/monitoring-custom-metrics/deployment/service-8080.yaml
+    
+    kubectl apply -f https://raw.githubusercontent.com/kyma-project/examples/master/monitoring-custom-metrics/deployment/service-8081.yaml
+
+    kubectl apply -f https://raw.githubusercontent.com/kyma-project/examples/master/monitoring-custom-metrics/deployment/service-monitor.yaml
+    ```
+
+    ```bash
+    kubectl get pods
+    NAME                             READY     STATUS    RESTARTS   AGE
+    sample-metrics-c9f998959-jd2fz   2/2       Running   0          2m
+    sample-metrics-c9f998959-kfbp8   2/2       Running   0          2m
+    sample-metrics-c9f998959-nnp2n   2/2       Running   0          2m
+    sample-metrics-c9f998959-vdnkn   2/2       Running   0          2m
+    ```
+
+2. Run the `port-forward` on the sample-metrics-8081 Service for port 8081 to check the metrics
+    ```bash
+    kubectl port-forward svc/sample-metrics-8081 8081:8081
+    ```
+    Then open a browser and access [`http://localhost:8081/metrics`](http://localhost:8081/metrics)
```

Comment:
Can we add more space between this sentence and the screenshot?

Reply:
added a line break